### ISSUE TITLE
Conditional output filter fails if a colon is used inside `:then=` or `:else=`

### ISSUE THREAD

**Issue by superspace** [modxcms/revolution#16198]
### Summary
Conditional output filter fails if a colon is used inside `:then=` or `:else=` on Linux Systems

### Step to reproduce
A conditional output modifier:
```
[[*id:eq=`1`:then=`Resource ID: 1`:else=`Resource ID: not 1`]]
```

### Observed behavior
The value of [[*id]] is displayed

If id is 1:
```
1
```
If id is not 1, the actual ID is displayed

There are some more observations: if there is no colon used in `then` and id is not 1, `else` is displayed correctly even with colon used inside. Unless you chain another modifier, then it will also fail.

### Expected behavior

If id is 1:
```
Resource ID: 1 
```
If id is not 1
```
Resource ID: not 1
```

### Environment
MODX 3.0.1
PHP 7.4.29 / 8.1 (FPM)
Apache 2.4
MariaDB 10.3.32
Linux 3.10.0-1160.49.1.el7.x86_64

The Bug doesn't show on MODX 2.8.* and so far only on Linux Systems. On local MAMP installation on MacOS everything works fine.

**Comment by halftrainedharry** [comment]
This is a known issue related to this line in the code:

https://github.com/modxcms/revolution/blob/cb6f56a19af818ad4d1d62dc0d31b9819aaa876f/core/src/Revolution/Filters/modInputFilter.php#L56

The regular expression was changed in #14458 to allow nested output filters, but causes a lot of problems.

Same issue in the MODX forum: https://community.modx.com/t/output-modifier-issue-in-modx-3-0-1-pl/5376
Related issues: #16044, #16186

---

This particular problem can probably be fixed by adding a backtick to the first character set (``[^:=`]`` instead of `[^:=]`), but that doesn't solve all the other issues with this new regex pattern.

**Comment by jmacdonald-spectacle** [comment]
My workaround is to use the html entity ``&colon;`` instead.

**Comment by rthrash** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/hiding-fields-does-not-work-correctly/5976/6

**Comment by Mark-H** [comment]
Should be fixed with #16288.

**Comment by rthrash** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/a-colon-in-a-resources-content-is-breaking-a-template/6127/4
